galley/pkg/server/server.go

```diff
@@ -210,16 +194,40 @@ func (s *Server) Close() error {
 		_ = s.listener.Close()
 	}
 
-	if s.livenessProbe != nil {
-		_ = s.livenessProbe.Close()
-	}
+	// final attempt to purge buffered logs
+	_ = log.Sync()
+
+	return nil
+}
+
+//RunServer start Galley Server mode
+func RunServer(sa *Args, printf, fatalf shared.FormatFn, livenessProbeController,
+	readinessProbeController probe.Controller) error {
+	printf("Galley started with\n%s", sa)
 
-	if s.readinessProbe != nil {
-		_ = s.readinessProbe.Close()
+	s, err := New(sa)
+	if err != nil {
+		fatalf("Unable to initialize Galley Server: %v", err)
 	}
 
-	// final attempt to purge buffered logs
-	_ = log.Sync()
+	serverLivenessProbe := probe.NewProbe()
+	serverLivenessProbe.SetAvailable(nil)
```

Comment:
The liveness probe should be set to available after the gRPC server is started inside server.Run().

Reply:
done